framework/src/modules/nft/endpoint.ts

```diff
@@ -153,30 +153,42 @@ export class NFTEndpoint extends BaseEndpoint {
 			return { collectionIDs: [] };
 		}
 
+		const isNFTSupported = await this.isNFTSupported(context);
+
+		if (!isNFTSupported.isNFTSupported) {
+			throw new Error('NFT is not supported');
+		}
+
 		const supportedNFTsData = await supportedNFTsStore.get(
 			context.getImmutableMethodContext(),
 			chainID,
 		);
 
-		return {
-			collectionIDs: supportedNFTsData.supportedCollectionIDArray.map(collection =>
-				collection.collectionID.toString('hex'),
-			),
-		};
+		if (supportedNFTsData.supportedCollectionIDArray.length) {
+			return {
+				collectionIDs: supportedNFTsData.supportedCollectionIDArray.map(collection =>
+					collection.collectionID.toString('hex'),
+				),
+			};
+		}
+
+		return { collectionIDs: [] };
 	}
 
-	public async collectionExists(
+	public async isCollectionIDSupported(
 		context: ModuleEndpointContext,
 	): Promise<{ collectionExists: boolean }> {
```

Comment:
Can the response please be changed to { collectionIDSupported: boolean } and corresponding `isCollectionIDSupportedResponseSchema` definition.